projects/media-viewer/src/lib/toolbar/sub-toolbar/sub-toolbar.component.ts

```diff
@@ -1,18 +1,29 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { ToolbarButtonVisibilityService } from '../toolbar-button-visibility.service';
 import { ToolbarEventService } from '../toolbar-event.service';
+import { Observable } from 'rxjs';
+import { select, Store } from '@ngrx/store';
+import * as fromSelectors from '../../store/selectors/icp.selectors';
+import { IcpSession, IcpState } from '../../icp/icp.interfaces';
 
 @Component({
   selector: 'mv-sub-toolbar',
   templateUrl: './sub-toolbar.component.html'
 })
-export class SubToolbarComponent {
+export class SubToolbarComponent implements OnInit {
+
+  icpSession$: Observable<IcpSession>;
 
   constructor(
     public readonly toolbarButtons: ToolbarButtonVisibilityService,
-    public readonly toolbarEvents: ToolbarEventService
+    public readonly toolbarEvents: ToolbarEventService,
+    private store: Store<IcpState>
   ) {}
 
+  ngOnInit() {
+    this.icpSession$ = this.store.pipe(select(fromSelectors.getIcpSession));
```

Comment:
we can just check the icpEnabled boolean from toolbar events. that's what we're doing with the other launch button